checks.d/postgres.py

```diff
@@ -292,6 +314,23 @@ def _get_bgw_metrics(self, key, db):
             metrics = self.bgw_metrics.get(key)
         return metrics
 
+    def _get_replication_metrics(self, key, db):
+        """ Use either REPLICATION_METRICS_9_1 or REPLICATION_METRICS_9_1 + REPLICATION_METRICS_9_2
+        depending on the postgres version.
+        Uses a dictionnary to save the result for each instance
+        """
+        metrics = self.replication_metrics.get(key)
+        if metrics is None:
+            # Only available for >= 9.2 due to
+            # pg_xlog_location_diff
+            self.replication_metrics[key] = dict(self.REPLICATION_METRICS_9_1)
```

Comment:
Could you put the `_is_9_1_or_above` test logic here and only keep the `_get_replication_metrics` in `_collect_stats` for more consistency?